### PR TITLE
adding sparse sensing postprocessor and associated plots

### LINKED ISSUE #2043: [TASK] Optimal sensor placement
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
Given some response of interest (i.e., temperature profile), it will be great if we can inform users of the optimal locations to place the thermocouples such that they can reconstruct the temperature profile with high precision.

**Describe alternatives you've considered**
N/A.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by Jimmy-INL**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #2043

##### What are the significant changes in functionality due to this change request?
It will inform the users of the optimal locations to place their sensors in order to reconstruct the responses of interest with high precision.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by Jimmy-INL** [comment]
@joshua-cogliati-inl, this PR has passed all checks. If you can assign a task to Niharika to populate the documentation to that part and create another PR to [Jimmy-INL:Jimmy-Josh-AddingSPSL-PP](https://github.com/Jimmy-INL/raven/tree/Jimmy-Josh-AddingSPSL-PP). Also, I tried to add a space to the 'Temperature (K)' as the label of the colorbar but it still misses the last round bracket. it labels it as `Temperature (K`

**Comment by joshua-cogliati-inl** [comment]
> @joshua-cogliati-inl, this PR has passed all checks. If you can assign a task to Niharika to populate the documentation to that part and create another PR to [Jimmy-INL:Jimmy-Josh-AddingSPSL-PP](https://github.com/Jimmy-INL/raven/tree/Jimmy-Josh-AddingSPSL-PP). Also, I tried to add a space to the 'Temperature (K)' as the label of the colorbar but it still misses the last round bracket. it labels it as `Temperature (K`

I am not quite sure what documentation you want created?

Hm, I wonder if somewhere the number of letters is being trimmed for the colorbar label?

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    if self.seed is not None:
+      model.fit(data, seed=self.seed)
+    else:
+      model.fit(data)
+    selectedSensors = model.get_selected_sensors()
+    coords = {'sensor':np.arange(1,len(selectedSensors)+1)}
+
+    sensorData = {}
+    for var in self.sensingFeatures:
+      sensorData[var] = ('sensor', inputDS[var][0,selectedSensors].data) #inputDS[self.sensingFeatures]
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:197]: it seems the comment at the end of this line is not related.

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    outDS = xr.Dataset(data_vars=sensorData, coords=coords)
+    ## PLEASE READ: For developers: this is really important, currently,
+    # you have to manually add RAVEN_sample_ID to the dims if you are using xarrays
+    outDS = outDS.expand_dims('RAVEN_sample_ID')
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:201]: Let's define a self.sampleTag = 'RAVEN_sample_ID' in __init__, and use it here.

File: tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>tests/framework/Postprocessors/SPSL</name>
+    <author>Mohammad Abdo (@Jimmy-INL)</author>
+    <created>2022-07-18</created>
+    <classesTested>Postprocessors.SPSL</classesTested>
+    <description>test SPSL reconstruction</description>
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml:8]: Please add more descriptions

File: tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          <y>outMC|Output|y</y>
+          <c>blue</c>
+        </plot>
+        <!-- <plot>
+          <type>scatter</type>
+          <marker>+</marker>
+          <x>romOut|Input|angle</x>
+          <y>romOut|Output|r</y>
+          <c>red</c>
+        </plot> -->
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml:134]: remove?

File: tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      <!-- <Input>v0,angle</Input> -->
+      <Output>x,y,v,t</Output>
+      <options>
+        <pivotParameter>t</pivotParameter>
+      </options>
+    </HistorySet>
+    <!-- <HistorySet name="romIn">
+      <Input>angle</Input>
+      <Output>OutputPlaceHolder</Output>
+    </HistorySet> -->
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml:103]: remove?

File: tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    </MonteCarlo>
+  </Samplers>
+
+  <DataObjects>
+    <PointSet name="inputPlaceHolder">
+      <Input>x0,v0,y0,angle</Input>
+      <Output>OutputPlaceHolder</Output>
+      <!-- <options>
+        <pivotParameter>t</pivotParameter>
+      </options> -->
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml:91]: remove?

File: tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml [hunk 7]
```diff
@@ -0,0 +1,154 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>tests/framework/Postprocessors/SPSL</name>
+    <author>Mohammad Abdo (@Jimmy-INL)</author>
+    <created>2022-05-24</created>
+    <classesTested>Postprocessors.SPSL</classesTested>
+    <description>test SPSL reconstruction</description>
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLReconstructioProjectile.xml:8]: Please update the description here, please add more details.

File: tests/framework/PostProcessors/SparseSensing/tests [hunk 8]
```diff
@@ -0,0 +1,8 @@
+[Tests]
+  [./testSPSLOptiTwist]
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/tests:2]: Two tests are added, but only one is added here, please check it.

**Review by wangcj05 — CHANGES_REQUESTED** [review]
@Jimmy-INL I have several comments for you to consider. In addition, are you able to reduce the the size of csv files, for example, using coarse mesh for the problem

File: tests/framework/PostProcessors/SparseSensing/tests [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+    type = 'RavenFramework'
+    input = 'testSPSLOptiTwist.xml'
+    csv = 'reconstructionOptiTwist/outPP.csv'
+    output = 'reconstructionOptiTwist/mySensorPlot_scatter-scatter.png'
```
Comment (Jimmy-INL) [inline tests/framework/PostProcessors/SparseSensing/tests:6]: @joshua-cogliati-inl, could you please modify this to test the png (compare with gold)? In addition, it seems it plots the last sample or maybe all of them on top of each other. Can we add user input to require which temperature (which sample) to be plotted? or at least plot the reference one?

Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/SparseSensing/tests:6]: FYI, if test the png, then we will only test when we test optional packages.

Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/SparseSensing/tests:6]: Hm, plotting a specific sample may be hard with the way the data is arranged.  For example, I tried:
```diff
diff --git a/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml 
b/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
index aa29c90d7..f8c28c01f 100644
--- a/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
+++ b/tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml
@@ -10,7 +10,7 @@
 
   <RunInfo>
     <WorkingDir>reconstructionOptiTwist</WorkingDir>
-    <Sequence>LoadCSV, mySPpp, print</Sequence>
+    <Sequence>LoadCSV, mySPpp,avetemp, print</Sequence>
     <batchSize>1</batchSize>
   </RunInfo>
 
@@ -28,10 +28,18 @@
       <Model class="Models" type="PostProcessor">mySPSL</Model>
       <Output class="DataObjects" type="DataSet">outPP</Output>
     </PostProcess>
+    <PostProcess name="avetemp">
+      <Input class="DataObjects" type="HistorySet">TwistDO</Input>
+      <Model class="Models" type="PostProcessor">average_temp</Model>
+      <Output class="DataObjects" type="PointSet">TwistDOTempAve</Output>
+    </PostProcess>
     <IOStep name="print">
       <Input class="DataObjects" type="DataSet">outPP</Input>
       <Input class="DataObjects" type="HistorySet">TwistDO</Input>
+      <Input class="DataObjects" type="PointSet">TwistDOTempAve</Input>
       <Output class="OutStreams" type="Print">outPP</Output>
+      <Output class="OutStreams" type="Print">outTwistDO</Output>
+      <Output class="OutStreams" type="Print">outTwistDOTempAve</Output>
       <Output class="OutStreams" type="Plot">mySensorPlot</Output>
     </IOStep>
   </Steps>
@@ -48,6 +56,10 @@
         <optimizer>QR</optimizer>
       </Goal>
     </PostProcessor>
+    <PostProcessor name="average_temp" subType="HStoPSOperator">
+      <pivotParameter>BCTemperature (K)</pivotParameter>
+      <operator>average</operator>
+    </PostProcessor>
   </Models>
 
   <DataObjects>
@@ -58,6 +70,10 @@
         <pivotParameter>index</pivotParameter>
       </options>
     </HistorySet>
+    <PointSet name="TwistDOTempAve">
+      <Input>Power (W)</Input>
+      <Output>X (m),Temperature (K),Y (m)</Output>
+    </PointSet>
     <DataSet name="outPP">
       <Input>InputPlaceHolder</Input>
       <Output>Temperature (K), X (m), Y (m)</Output>
@@ -71,6 +87,14 @@
       <source>outPP</source>
       <what>output</what>
     </Print>
+    <Print name="outTwistDO">
+      <type>csv</type>
+      <source>TwistDO</source>
+    </Print>
+    <Print name="outTwistDOTempAve">
+      <type>csv</type>
+      <source>TwistDOTempAve</source>
+    </Print>
     <Plot name="mySensorPlot">
       <plotSettings>
         <plot>
```
but that resulted in:
```
Power (W),X (m),Temperature (K),Y (m)
350,0.0107616743047,241.831291133,0.08204888554
350,0.0107616743047,273.515621667,0.08204888554
350,0.0107616743047,303.282704721,0.08204888554
350,0.0107616743047,332.908743084,0.08204888554
```
which drops the index. I am not sure there is a quick fix for this.

Comment (Jimmy-INL) [reply tests/framework/PostProcessors/SparseSensing/tests:6]: If we repeat the test multiple times the way it was, is the color map consistent? which temp is it showing? all on top of each other, or is it the first, or last? can you check that?

Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/SparseSensing/tests:6]: Based on the color map, it is showing the 330 K BC temperature data.

Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/SparseSensing/tests:6]: 3dsensorplot
Here is a plot with:
```
        <plot>
          <type>scatter</type>
          <x>TwistDO|Output|X (m)</x>
          <y>TwistDO|Output|Y (m)</y>
          <z>TwistDO|Input|BCTemperature (K)</z>
          <cmap>coolwarm</cmap>
          <colorMap>TwistDO|Output|Temperature (K)</colorMap>
        </plot>
```
So the data for all four temperatures is there, but just the highest boundary condition is drawn.

File: ravenframework/utils/InputData.py [hunk 10]
```diff
@@ -534,6 +534,9 @@ def returnDefault(self, name):
     sub = self.getSub(name)
     if sub is not None:
       return sub.default
+    else:
+      pass #uncomment next line if needed for debugging.
+      #raise ValueError("Unrecognized variable '{}'".format(name))
```
Comment (joshua-cogliati-inl) [inline ravenframework/utils/InputData.py:539]: Should these lines stay in for debugging, or can the be removed?  (FYI: the docstring for the function says that this returns None, not raise an error for subs that are not found.)

File: doc/user_manual/PostProcessors/SparseSensing.tex [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    \item Brunton, Bingni W., et al. ``Sparse sensor placement optimization for classification.'' SIAM Journal on Applied Mathematics 76.5 (2016): 2099-2122 for classification and,
+    \item de Silva, Brian M., et al. ``PySensors: A Python package for sparse sensor placement.'' arXiv preprint arXiv:2102.13476 (2021) contains a full literature review along with examples and additional tips for using PySensors effectively.
+\end{itemize}
+
+\ppType{SparseSensing }{SparseSensing}
+\begin{itemize}
+\item \xmlNode{Goal}, \xmlDesc{string, required field}, the goal of the sparse sensor optimization.
+User has to provide \xmlAttr{subType} which is a \xmlDesc{string, required field} representing the goal of the sparse sensing optimization; i.e., which goal function is used in the optimization? Examples for such goal functions are:
+\begin{itemize}
+    \item \textbf{Reconstruction} deals with predicting the values of a quantity of interest at different locations other than those where sensors are located. For example, one might predict the temperature at a point point in the middle of a fuel rod based on readings taken at various other positions.
```
Comment (wangcj05) [inline doc/user_manual/PostProcessors/SparseSensing.tex:20]: remove duplicated 'point'?

File: doc/user_manual/PostProcessors/SparseSensing.tex [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+    \item \textbf{Classification} is the problem of predicting which category an example belongs to, given a set of training data (e.g. determining whether digital photos are of dogs or cats).
+\end{itemize}
+\nb{Currently, only reconstruction is implemented.}
+In order to use the \xmlNode{Goal}, the user needs to provide the following subnodes:
+\begin{itemize}
+	\item \xmlNode{features}, \xmlDesc{comma separated strings, required field}, features/inputs of the data model, i.e., locations or time stamps we should sense
+	\item \xmlNode{target}, \xmlDesc{comma separated strings, required field}, target of data model
```
Comment (wangcj05) [inline doc/user_manual/PostProcessors/SparseSensing.tex:28]: I would prefer either "feature, target" pair or "features, targets" pair.

File: doc/user_manual/PostProcessors/SparseSensing.tex [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+	\item \xmlNode{basis} , \xmlDesc{string, optional field}, the type of basis onto which the data are projected: \xmlString{Identity},\xmlString{ SVD}, \xmlString{Random}. \default{SVD}
```
Comment (wangcj05) [inline doc/user_manual/PostProcessors/SparseSensing.tex:29]: remove extra space in '\xmlString{ SVD}'?

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+    goal = InputData.parameterInputFactory('Goal',
+                                                  printPriority=108,
+                                                  descr=r"""The goal of the sparse sensor optimization (i.e., reconstruction or classification)""")
+    goal.addParam("subType", InputTypes.makeEnumType("Goal", "GoalType", ['reconstruction','classification']), False, default='reconstruction')
+    inputSpecification.addSub(goal)
+    features = InputData.parameterInputFactory("features", contentType=InputTypes.StringListType,
+                                                printPriority=108,
+                                                descr=r"""Features/inputs of the data model""")
+    goal.addSub(features)
+    target = InputData.parameterInputFactory("target", contentType=InputTypes.StringType,
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:55]: See my previous comment about feature/target

File: doc/user_manual/PostProcessors/SparseSensing.tex [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+	\item \xmlNode{nModes},  \xmlDesc{integer, required field}, the number of modes  used to project the data.
+	\item \xmlNode{nSensors},  \xmlDesc{integer, required field}, the number of sensors used
+	\item \xmlNode{optimizer},  \xmlDesc{string, optional field}, the optimizer used to find the sensors: \xmlString{QR}, \xmlString{CCQR} for unconstrained and cost constrained respectively.
```
Comment (wangcj05) [inline doc/user_manual/PostProcessors/SparseSensing.tex:32]: Could you update this description, only 'QR' is used inside the code right now.

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+                                                           descr=r"""The number of sensors used""")
+    goal.addSub(nSensors)
+    optimizer = InputData.parameterInputFactory("optimizer", contentType=InputTypes.makeEnumType("optimizer","optimizer type",['QR']),
+                                                           printPriority=108,
+                                                           descr=r"""The type of optimizer used""",default='QR')
+    goal.addSub(optimizer)
+    seed = InputData.parameterInputFactory("seed", contentType=InputTypes.IntegerType,
+                                                           printPriority=108,
+                                                           descr=r"""The integer seed use for sensor placement random number seed""")
+    goal.addSub(seed)
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:78]: It seems the 'seed' keyword is not explained in the user manual.

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+      self.optimizer = child.findFirst('optimizer').value
+      if child.findFirst('seed') is not None:
+        self.seed = child.findFirst('seed').value
+      else:
+        self.seed = None
+      if child.parameterValues['subType'] not in self.goalsDict.keys():
+        self.raiseAnError(IOError, '{} is not a recognized option, allowed options are {}'.format(child.getName(),self.goalsDict.keys()))
+    _, notFound = paramInput.subparts[0].findNodesAndExtractValues(['nModes','nSensors','features','target'])
+    # notFound must be empty
+    assert(not notFound)
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:138]: I think you need to remove the bracket for the assert, otherwise it is always true.

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+
+  def run(self,inputIn):
+    """
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:141]: Please add some description for the run method.

File: ravenframework/Models/PostProcessors/SparseSensing.py [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+      self.raiseAnError(IOError, 'optimizer {} not implemented!!!'.format(self.optimizer))
+
+    model = ps.SSPOR(basis=basis,n_sensors = self.nSensors,optimizer = optimizer)
+
+    features = {}
+    for var in self.sensingFeatures:
+      features[var] = np.atleast_1d(inputDS[var].data)
+
+    data = inputDS[self.sensingTarget].data
+    ## TODO: add some assertions to check the shape of the data matrix in case of steady state and time-dependent data
```
Comment (wangcj05) [inline ravenframework/Models/PostProcessors/SparseSensing.py:173]: Can you check the checks in the "TODO"?

File: ravenframework/utils/InputData.py (hunk 10 above)
Comment (wangcj05) [reply ravenframework/utils/InputData.py:539]: @Jimmy-INL Could you provide your answer for Josh's comment here?

File: tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml [hunk 20]
```diff
@@ -0,0 +1,128 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <TestInfo>
+    <name>tests/framework/Postprocessors/SPSL</name>
+    <author>Mohammad Abdo (@Jimmy-INL)</author>
+    <created>2022-07-18</created>
+    <classesTested>Postprocessors.SPSL</classesTested>
```
Comment (wangcj05) [inline tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml:7]: Please change SPSL to SparseSensing

**Review by wangcj05 — CHANGES_REQUESTED** [review]
@Jimmy-INL I have a couple of comments for you to consider. In addition, is it possible for you to reduce the size of CSV files?

File: ravenframework/Models/PostProcessors/SparseSensing.py (hunk 17 above)
Comment (joshua-cogliati-inl) [reply ravenframework/Models/PostProcessors/SparseSensing.py:138]: I am not sure what you are asking here? Which bracket?

```
>>> assert(not True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> assert(not False)
>>> 
```

File: tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml (hunk 20 above)
Comment (joshua-cogliati-inl) [reply tests/framework/PostProcessors/SparseSensing/testSPSLOptiTwist.xml:7]: Fixed.

File: ravenframework/Models/PostProcessors/SparseSensing.py (hunk 17 above)
Comment (joshua-cogliati-inl) [reply ravenframework/Models/PostProcessors/SparseSensing.py:138]: I changed it to:
```
assert not notFound, "Unexpected nodes in _handleInput"
```

File: doc/user_manual/PostProcessors/SparseSensing.tex (hunk 13 above)
Comment (joshua-cogliati-inl) [reply doc/user_manual/PostProcessors/SparseSensing.tex:29]: Fixed.

File: doc/user_manual/PostProcessors/SparseSensing.tex (hunk 11 above)
Comment (joshua-cogliati-inl) [reply doc/user_manual/PostProcessors/SparseSensing.tex:20]: Fixed.

File: ravenframework/utils/InputData.py (hunk 10 above)
Comment (Jimmy-INL) [reply ravenframework/utils/InputData.py:539]: These lines were actually added by @joshua-cogliati-inl. I have no issues with removing them. @joshua-cogliati-inl, please confirm that we do not need them and I will remove them.

File: ravenframework/Models/PostProcessors/SparseSensing.py (hunk 19 above)
Comment (Jimmy-INL) [reply ravenframework/Models/PostProcessors/SparseSensing.py:175]: @wangcj05, this is for the case of time-dependent data, which is not done yet. I left this as a reminder for Josh when he implements it.

I have added another assert here for the steady state case, yet I don't think it is possible to make it false.

File: ravenframework/utils/InputData.py (hunk 10 above)
Comment (joshua-cogliati-inl) [reply ravenframework/utils/InputData.py:539]: Yes, we can remove them.

File: ravenframework/Models/PostProcessors/SparseSensing.py (hunk 16 above)
Comment (joshua-cogliati-inl) [reply ravenframework/Models/PostProcessors/SparseSensing.py]: We decided it was not needed.  (The pysensors documentation on it is at: https://python-sensors.readthedocs.io/en/latest/api/pysensors.html#pysensors.SSPOR.fit )

Comment (joshua-cogliati-inl) [reply ravenframework/Models/PostProcessors/SparseSensing.py:78]: We documented the seed parameter.

**Comment by wangcj05** [comment]
@Jimmy-INL @joshua-cogliati-inl It seems some of my comments have not yet addressed, could you check for that? In addition, is it possible to reduce the size of csv files?

**Comment by joshua-cogliati-inl** [comment]
> @Jimmy-INL @joshua-cogliati-inl It seems some of my comments have not yet addressed, could you check for that? In addition, is it possible to reduce the size of csv files?

I pushed up a version that only kept every 10th line, so they are now smaller.

File: doc/user_manual/PostProcessors/SparseSensing.tex (hunk 12 above)
Comment (joshua-cogliati-inl) [reply doc/user_manual/PostProcessors/SparseSensing.tex:28]: @Jimmy-INL How should we resolve this?

**Review by joshua-cogliati-inl — COMMENTED** [review]
I have reviewed it and I think it is ready for merging.

**Comment by wangcj05** [comment]
checklist is good, PR can be merged.
